File: app/src/main/java/com/bignerdranch/android/mediafiles/ui/home/HomeFragment.java
package com.bignerdranch.android.mediafiles.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bignerdranch.android.mediafiles.R;
import com.bignerdranch.android.mediafiles.ui.recycleView.MediaFileAdapter;

public class HomeFragment extends Fragment {

    private HomeViewModel homeViewModel;
    private MediaFileAdapter adapter;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        homeViewModel =
                ViewModelProviders.of(this).get(HomeViewModel.class);
        View root = inflater.inflate(R.layout.fragment_home, container, false);
        final TextView textView = root.findViewById(R.id.text_summary);

        final LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        layoutManager.setOrientation(RecyclerView.VERTICAL);
        final RecyclerView localItemRecyclerView = root.findViewById(R.id.mediaFilesRecyclerView);
        localItemRecyclerView.setLayoutManager(layoutManager);
        adapter = new MediaFileAdapter();
        localItemRecyclerView.setAdapter(adapter);

        homeViewModel.getLiveMediaFiles().observe(getViewLifecycleOwner(), adapter::submitList);
        homeViewModel.getText().observe(getViewLifecycleOwner(), new Observer<String>() {
            @Override
            public void onChanged(@Nullable String s) {
                textView.setText(s);
            }
        });

        return root;
    }
}

File: app/src/main/java/com/bignerdranch/android/mediafiles/ui/home/HomeFragment.kt
package com.bignerdranch.android.mediafiles.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.repeatOnLifecycle
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.bignerdranch.android.mediafiles.MediaFilesApplication
import com.bignerdranch.android.mediafiles.databinding.FragmentHomeBinding
import com.bignerdranch.android.mediafiles.ui.recycleView.MediaFileAdapter
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class HomeFragment : Fragment() {
    lateinit var homeViewModel: HomeViewModel
    lateinit var adapter: MediaFileAdapter

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val binding = FragmentHomeBinding.inflate(layoutInflater)

        homeViewModel = ViewModelProvider(this).get(HomeViewModel::class.java)
        adapter = MediaFileAdapter(MediaFilesApplication.appComponent.getLogger())

        binding.bindAdapter(adapter)

        //homeViewModel.liveMediaFiles.observe(viewLifecycleOwner, Observer { pagedList: PagedList<MediaFile> -> adapter.submitList(pagedList) })
        lifecycleScope.launch {
            // We repeat on the STARTED lifecycle because an Activity may be PAUSED
            // but still visible on the screen, for example in a multi window app
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                homeViewModel.liveMediaFiles.collectLatest {
                    adapter.submitData(it)
                }
            }
        }

        homeViewModel.text.observe(viewLifecycleOwner, Observer { s -> binding.textSummary.text = s })
        return binding.root
    }

    /**
     * Sets up the [RecyclerView] and binds [MediaFileAdapter] to it
     */
    private fun FragmentHomeBinding.bindAdapter(mediaFileAdapter: MediaFileAdapter) {
        mediaFilesRecyclerView.adapter = mediaFileAdapter
        val layoutManager = LinearLayoutManager(context)
        layoutManager.orientation = RecyclerView.VERTICAL
        mediaFilesRecyclerView.layoutManager = layoutManager
        val decoration = DividerItemDecoration(mediaFilesRecyclerView.context, DividerItemDecoration.VERTICAL)
        mediaFilesRecyclerView.addItemDecoration(decoration)
    }

}